pinot-segment-local/src/main/java/org/apache/pinot/segment/local/indexsegment/immutable/ImmutableSegmentImpl.java

```diff
@@ -102,6 +102,10 @@ public void enableUpsert(PartitionUpsertMetadataManager partitionUpsertMetadataM
     _validDocIds = validDocIds;
   }
 
+  public void setValidDocIds(ThreadSafeMutableRoaringBitmap validDocIds) {
```

Comment:
Suggest not adding this method since we want to keep a reference to the upsert metadata manager